vars/updatecli.groovy

```diff
@@ -1,58 +1,95 @@
+    // Set cron trigger if requested
+    if (finalConfig.cronTriggerExpression) {
+        properties([pipelineTriggers([cron(finalConfig.cronTriggerExpression)])])
+    }
 
-  def updatecliCommand = 'updatecli ' +  finalConfig.action
-  // Do not add the flag "--config" if the provided value is "empty string"
-  updatecliCommand += finalConfig.config ? " --config ${finalConfig.config}" : ''
-  // Do not add the flag "--values" if the provided value is "empty string"
-  updatecliCommand += finalConfig.values ? " --values ${finalConfig.values}" : ''
+    node(finalConfig.updatecliAgentLabel) {
+        final String updatecliRunStage = "Run updatecli: ${finalConfig.action}"
+        boolean runUpdatecli = true
 
-  // Define a cron trigger only if requested by the user through attribute
-  if (finalConfig.cronTriggerExpression) {
-    properties([pipelineTriggers([cron(finalConfig.cronTriggerExpression)])])
-  }
+        stage("Check if ${finalConfig.config} folder exists: ${finalConfig.action}") {
+            checkout scm
+            if (!fileExists(finalConfig.config)) {
+                echo "WARNING: no ${finalConfig.config} folder."
+                runUpdatecli = false
+                org.jenkinsci.plugins.pipeline.modeldefinition.Utils.markStageSkippedForConditional(updatecliRunStage)
+            }
+        }
 
+        stage(updatecliRunStage) {
+            withEnv(["PATH+CUSTOM=$customUpdatecliPath"]) { // Wrap entire execution inside PATH update
+                if (runUpdatecli) {
+                    // If a custom version is provided, download and install that version of updatecli
+                    if (finalConfig.version) {
+                        stage("Download updatecli version ${finalConfig.version}") {
+                            sh 'echo $PATH'
+                            withEnv(["UPDATECLI_VERSION=${finalConfig.version}", "CUSTOM_UPDATECLI_PATH=${customUpdatecliPath}"]) { // Only custom installation vars
+                                sh '''
+                                    versionTag="v${UPDATECLI_VERSION}"
+                                    cpu="$(uname -m)"
+                                    # Normalize CPU architecture for Updatecli release filenames
+                                    if [ "$cpu" = "aarch64" ] || [ "$cpu" = "arm64" ]; then
+                                        cpu="arm64"
+                                    fi
+                                    # Construct the tar file name based on the CPU architecture.
+                                    tarFileName="updatecli_Linux_${cpu}.tar.gz"
+                                    downloadUrl="https://github.com/updatecli/updatecli/releases/download/${versionTag}/${tarFileName}"
+                                    echo "Downloading updatecli version ${UPDATECLI_VERSION} from ${downloadUrl}"
+                                    curl --silent --location --output ${tarFileName} ${downloadUrl} || { echo "Download failed"; exit 1; }
```

Comment:
```suggestion
                                    curl --silent --show-error --location --output ${tarFileName} ${downloadUrl}
```

- Curl flag `--show-error`: Prints HTTP or network error if any: https://curl.se/docs/manpage.html#-S
- Removed the `|| xxxx` manual error handling as it is managed by the shell (with an implicit flag `set -e` added by Jenkins)